include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+        $calendarList = $this->gService->calendarList->listCalendarList();
+
+        // find the id of the 'SuiteCRM' calendar ... in the future, this will set the calendar of the users choosing.
+        $this->calendarId = $this->getSuiteCRMCalendar($calendarList);
+
+        // if the SuiteCRM calendar doesn't exist... Create it!
+        if (!$this->isCalendarExists()) {
+            // TODO: It should be an exception?
+            $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to find the SuiteCRM Google Calendar, Creating it!');
+            $calendar = new Google_Service_Calendar_Calendar();
+            $calendar->setSummary('SuiteCRM');
+            $calendar->setTimeZone($this->timezone);
+
+            $createdCalendar = $this->gService->calendars->insert($calendar);
+            $this->calendarId = $createdCalendar->getId();
+        }
+
+        // Final check to make sure we have an ID
+        if (!$this->isCalendarExists()) {
+            return false;
+        }
+        return $this->calendarId;
+    }
+
+    /**
+     * find the id of the 'SuiteCRM' calendar ... in the future, this will return the calendar of the users choosing.
+     *
+     * @param Google_Service_Calendar_CalendarList $calendarList
+     * 
+     * @return string|null Matching Google Calendar ID or null.
+     */
+    protected function getSuiteCRMCalendar(Google_Service_Calendar_CalendarList $calendarList)
+    {
+        foreach ($calendarList->getItems() as $calendarListEntry) {
+            if ($calendarListEntry->getSummary() == 'SuiteCRM') {
+                $return = $calendarListEntry->getId();
+                break;
+            }
+        }
+        if (empty($return)) { // TODO: this empty-check is not neccessary, return the getId() results as it is.
+            return null;
+        }
+        return $return;
+    }
+
+    /**
+     * Get events in users google calendar
+     *
+     *
+     * @return bool|array Array of Google_Service_Calendar_Event Objects
+     */
+    protected function getUserGoogleEvents()
+    {
+
+        // Make sure we have a Google Calendar Service instance and
+        // Make sure we have a calendar id
+        if (!$this->isServiceExists() || !$this->isCalendarExists()) {
+            return false;
+        }
+
+        // Set Options for what events we get from Google
+        $optParams = array(
+            'maxResults' => 500,
+            'showDeleted' => true,
+            'singleEvents' => true,
+            'timeMin' => date('c', strtotime('-1 month')),
+            'timeMax' => date('c', strtotime('+3 month')),
+        );
+
+        $results_g = $this->gService->events->listEvents($this->calendarId, $optParams);
+
+        // We only want the events, not the leading cruft
+        $results = $results_g->getItems(); // TODO: getItems() returns a Google_Service_Calendar_Event, but this code expect an array in the next line? Something missing here?
```

Comment:
@gymad
listEvents() above returns Google_Service_Calendar_Events
getItems() returns an array of Google_Service_Calendar_Event objects.
Google made the names a bit confusing.